src/js/utils/strings.js

```diff
@@ -99,6 +99,17 @@ export function seconds(str, frameRate) {
     return sec;
 }
 
+// Convert an offset string to a number; supports conversion of percentage offsets
+export function offsetToSeconds(offset, duration, frameRate) {
+    if (_.isString(offset) && offset.slice(-1) === '%') {
+        const percent = parseFloat(offset);
+        if (isNaN(duration) || isNaN(percent)) {
```

Comment:
If duration is `0` (or falsy), then we also do not want to return a final result:

`if (!duration || isNaN(duration) || isNaN(percent)) {`